### PR TITLE
fix: testLoadRules

### LINKED ISSUE #14: The `testLoadRules` fails depending on the order of execution
The test `testLoadRules` in the `MappingRulesLoaderTest` class can fail depending on the order when it's executed.
If it runs first everything works fine, but if it runs after the `testInitMoreThanOnce`, it fails.
This is because the usage of singleton in `MappingRulesLoader` which is of course "shared" across tests.
If `testInitMoreThanOnce` runs first, it initialize the loader so the `testLoadRules` will fail.

### PR DESCRIPTION
**Pull request by antonio-pedro99**
This PR fixes #14 

Signed-off-by: Antonio Pedro <tonio.pedro99@gmail.com>

### CODE REVIEW

File: src/main/java/io/strimzi/kafka/bridge/mqtt/utils/MappingRulesLoader.java [hunk 1]
```diff
@@ -46,7 +46,10 @@ private MappingRulesLoader() {
      *
      * @return the singleton instance of the MappingRulesLoader
      */
-    public static MappingRulesLoader getInstance() {
+    public static MappingRulesLoader getInstance() throws NullPointerException {
```
Comment (AlexJuca) [inline src/main/java/io/strimzi/kafka/bridge/mqtt/utils/MappingRulesLoader.java:49]: @antonio-pedro99 Should thread safety be a consideration here?

In its current implementation, multiple threads can potentially access the getInstance() method simultaneously and create multiple instances of the MappingRulesLoader class, which contradicts the intention of having a singleton pattern. If this is not a concern, kindly ignore this. 

For more info on the topic check out this article:

https://docs.oracle.com/javase/tutorial/essential/concurrency/sync.html

Comment (antonio-pedro99) [reply src/main/java/io/strimzi/kafka/bridge/mqtt/utils/MappingRulesLoader.java:49]: Yeap, you are right!
Thanks for point it out @AlexJuca.

**Review by ppatierno — COMMENTED** [review]
I don't think this is the right way to go for testing a singleton class. Just adding a close method for testing reveals an antipattern for singleton, when usually you do not need to close it across the application but just having one instance forever. 
The way to go for testing should be mocking the singleton for each test. You could use mockito to do that.

**Comment by antonio-pedro99** [comment]
@ppatierno thanks for pointing down. 
I am on it.

**Comment by antonio-pedro99** [comment]
Closed as the recent #17 fixes this issue.

**Pull request closed without merging**